tests/test_grpc.py

```diff
@@ -373,3 +374,31 @@ def test_end_training_denied(  # pylint: disable=unused-argument
     with pytest.raises(grpc.RpcError):
         reply = participant_stub.EndTraining(coordinator_pb2.EndTrainingRequest())
         assert reply.status_code == grpc.StatusCode.PERMISSION_DENIED
+
+
+@pytest.mark.integration
+def test_start_participant(mock_coordinator_service):
+    """[summary]
+
+    .. todo:: Advance docstrings (https://xainag.atlassian.net/browse/XP-425)
+
+    Args:
+        mock_coordinator_service ([type]): [description]
+    """
+
+    init_weight = [np.arange(10), np.arange(10, 20)]
+    mock_coordinator_service.coordinator.weights = init_weight
+
+    # mock a local participant with a constant train_round function
+    with mock.patch("xain_sdk.participant.Participant") as mock_obj:
```

Comment:
the object should be mocked where it is called instead of where it is defined, that means
`with mock.patch("xain_sdk.participant_state_machine.Participant") as mock_obj:`

Reply:
I didn't realise that. Is it just coincidence that "xain_sdk.participant.Participant" works just as well?

Reply:
i think in many cases it works coincidentally.

but it can also go wrong, yielding errors which look pretty strange on the first look, e.g. i had to change sth in the xain-proto repo in one of the last tickets, then merged the changes but couldn't publish it on pypi. ofcourse i still needed the changes in xain-proto for the followup tickets for xain-fl/sdk, therefor i adjusted their setup.py to reference to the dev-branch instead of the pypi package. so far so good, but then the mocking of one previously passing test failed, because it pointed to the xain-proto source for mocking instead of the xain-sdk's imported usage of the xain-proto stuff. long story short, as far as i have seen, it can happen that mocking the source can point to the "wrong" source, therefor the mocking happened at the source, but had no effect at the actual execution and the test/mocking failed.

Reply:
thanks for sharing this insight!